lnbits/extensions/boltz/crud.py

```diff
@@ -172,54 +172,162 @@ async def create_reverse_submarine_swap(
         VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
         """,
         (
-            swap.id,
+            reverse_swap.id,
+            reverse_swap.wallet,
+            reverse_swap.status,
+            reverse_swap.boltz_id,
+            reverse_swap.instant_settlement,
+            reverse_swap.preimage,
+            reverse_swap.claim_privkey,
+            reverse_swap.lockup_address,
+            reverse_swap.invoice,
+            reverse_swap.onchain_amount,
+            reverse_swap.onchain_address,
+            reverse_swap.timeout_block_height,
+            reverse_swap.redeem_script,
+            reverse_swap.amount,
+        ),
+    )
+
+    claim_task = asyncio.create_task(
+        client.claim_reverse_swap(
+            privkey_wif=claim_privkey_wif,
+            preimage_hex=preimage_hex,
+            lockup_address=swap.lockupAddress,
+            receive_address=data.onchain_address,
+            redeem_script_hex=swap.redeemScript,
+        )
+    )
+
+    pay_task = pay_invoice_and_update_status(
+        swap_id,
+        claim_task,
+        pay_invoice(
+            wallet_id=data.wallet,
+            payment_request=swap.invoice,
+            description=f"reverse swap for {swap.onchainAmount} sats on boltz.exchange",
+            extra={"tag": "boltz", "swap_id": swap_id, "reverse": True},
+        ),
+    )
+
+    asyncio.gather(claim_task, pay_task)
```

Comment:
This should really be in another file, not in `crud.py`. I would put as little logic into this file as possible and limit to only db operaitons.